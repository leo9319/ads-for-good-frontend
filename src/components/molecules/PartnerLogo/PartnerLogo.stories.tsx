import React from 'react';
import PartnerLogo from './PartnerLogo';
import { Meta, StoryObj } from '@storybook/react/*';

const meta = {
  title: 'Library/Molecules/PartnerLogo',
  component: PartnerLogo,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    className: {
      control: 'text',
    },
  },
  decorators: [
    Story => (
      <div style={{ padding: '20px' }}>
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof PartnerLogo>;

export default meta;

type Story = StoryObj<typeof meta>;

export const partnerIcon: Story = {
  args: {},
};
