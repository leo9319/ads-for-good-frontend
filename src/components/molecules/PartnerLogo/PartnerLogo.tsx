import React from 'react';
import FallbackImage from '@assets/images/FallbackImage.svg';
import ImageGroup from '@components/molecules/ImageGroup';

export type PartnerLogoProps = {
  /**
   * className for PartnerLogo container
   */
  className?: string;
};

export const PartnerLogo = (props: PartnerLogoProps) => {
  const partnerLogos = [
    {
      src: '',
      icon: 'icon-content-accredited-member',
      alt: 'p1',
      fallbackSrc: FallbackImage,
      height: '40px',
      width: '126px',
      link: {
        href: 'https://giveconfidently.ca/accredited-member/world-vision-canada-vision-mondiale-canada/',
        target: '_blank',
        name: 'Accredited Member - World Vision Canada',
      },
    },
    {
      src: '',
      icon: 'icon-content-bbb',
      alt: 'p2',
      fallbackSrc: FallbackImage,
      height: '40px',
      width: '102px',
      link: {
        href: 'https://www.bbb.org/',
        target: '_blank',
        name: 'Better Business Bureau',
      },
    },
    {
      src: '',
      icon: 'icon-content-imagine-canada',
      alt: 'p3',
      fallbackSrc: FallbackImage,
      height: '40px',
      width: '135px',
      link: {
        href: 'https://www.imaginecanada.ca/en',
        target: '_blank',
        name: 'Imagine Canada',
      },
    },
    {
      src: '',
      icon: 'icon-content-hc',
      alt: 'p4',
      fallbackSrc: FallbackImage,
      height: '40px',
      width: '135px',
      link: {
        href: 'https://www.worldvision.ca/humanitarian-coalition',
        target: '_blank',
        name: 'Humanitarian Coalition',
      },
    },
  ];

  return <ImageGroup className={props.className} data={partnerLogos} />;
};

export default PartnerLogo;
