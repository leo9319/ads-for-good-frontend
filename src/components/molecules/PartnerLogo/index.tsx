export type { PartnerLogoProps } from './PartnerLogo';
export { default } from './PartnerLogo';
export { PartnerLogo } from './PartnerLogo';
