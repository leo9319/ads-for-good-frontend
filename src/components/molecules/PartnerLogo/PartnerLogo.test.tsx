import React from 'react';
import PartnerLogo from './PartnerLogo';
import { render } from '@testing-library/react';

describe('PartnerLogo', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<PartnerLogo />);
    expect(baseElement).toMatchSnapshot();
  });
});
