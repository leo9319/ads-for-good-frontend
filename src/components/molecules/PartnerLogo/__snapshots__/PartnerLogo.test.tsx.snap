// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerLogo should render successfully 1`] = `
<body>
  <div>
    <div
      class="imageGroupContainer"
    >
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://giveconfidently.ca/accredited-member/world-vision-canada-vision-mondiale-canada/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p1"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper icon icon-content-accredited-member"
          inert=""
          src=""
          style="--imgWidth: 126px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.bbb.org/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p2"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper icon icon-content-bbb"
          inert=""
          src=""
          style="--imgWidth: 102px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.imaginecanada.ca/en"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p3"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper icon icon-content-imagine-canada"
          inert=""
          src=""
          style="--imgWidth: 135px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.worldvision.ca/humanitarian-coalition"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p4"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper icon icon-content-hc"
          inert=""
          src=""
          style="--imgWidth: 135px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
    </div>
  </div>
</body>
`;
